hivemind/utils/mpfuture.py

```diff
@@ -335,9 +288,8 @@ def __del__(self):
 
     def __getstate__(self):
         return dict(
-            synchronize=self.synchronize,
             _sender_pipe=self._sender_pipe,
-            _state=self._state,
+            _shared_state_code=self._shared_state_code,
```

Comment:
Pickling does not seem to serialize/deserialize `_shared_state_code` in a way that it is still shared with the original byte.

Reply:
Everything you said is correct, but we are not using pickle for serialization. Instead, MPFutures are passed through ForkingPickler: a serializer that is meant for passing data between processes spawned with fork.